kernel/h1/src/globalsec.rs

```diff
@@ -0,0 +1,220 @@
+        (0x000c => cpu0_d_region3_ctrl: ReadWrite<u32, REGION_CTRL::Register>),
+
+        (0x0010 => _reserved0010),
+
+        (0x0080 => ddma0_region0_ctrl: ReadWrite<u32, REGION_CTRL::Register>),
+        (0x0084 => ddma0_region1_ctrl: ReadWrite<u32, REGION_CTRL::Register>),
+        (0x0088 => ddma0_region2_ctrl: ReadWrite<u32, REGION_CTRL::Register>),
+        (0x008c => ddma0_region3_ctrl: ReadWrite<u32, REGION_CTRL::Register>),
+
+        (0x0090 => _reserved0090),
+
+        (0x00c0 => dusb0_region0_ctrl: ReadWrite<u32, REGION_CTRL::Register>),
+        (0x00c4 => dusb0_region1_ctrl: ReadWrite<u32, REGION_CTRL::Register>),
+        (0x00c8 => dusb0_region2_ctrl: ReadWrite<u32, REGION_CTRL::Register>),
+        (0x00cc => dusb0_region3_ctrl: ReadWrite<u32, REGION_CTRL::Register>),
+
+        (0x00d0 => _reserved00d0),
+
+        (0x00e0 => flash_region0_ctrl: ReadWrite<u32, REGION_CTRL::Register>),
+        (0x00e4 => flash_region1_ctrl: ReadWrite<u32, REGION_CTRL::Register>),
+        (0x00e8 => flash_region2_ctrl: ReadWrite<u32, REGION_CTRL::Register>),
+        (0x00ec => flash_region3_ctrl: ReadWrite<u32, REGION_CTRL::Register>),
+
+        (0x00f0 => _reserved00f0),
+
+        (0x0230 => flash_region0_base_addr: ReadWrite<u32>),
+        (0x0234 => flash_region0_size: ReadWrite<u32>),
+        (0x0238 => flash_region1_base_addr: ReadWrite<u32>),
+        (0x023c => flash_region1_size: ReadWrite<u32>),
+        (0x0240 => flash_region2_base_addr: ReadWrite<u32>),
+        (0x0244 => flash_region2_size: ReadWrite<u32>),
+        (0x0248 => flash_region3_base_addr: ReadWrite<u32>),
+        (0x024c => flash_region3_size: ReadWrite<u32>),
+
+        (0x0250 => @END),
+    }
+}
+
+register_bitfields![u32,
+    REGION_CTRL [
+        /// Enable for region. 0 means region accepts no transactions
+        EN OFFSET(0) NUMBITS(1) [],
+        /// Read enable for region. 1 means region accepts read transaction if EN is 1
+        RD_EN OFFSET(1) NUMBITS(1) [],
+        /// Write enable for region. 1 means region accepts write transaction if EN is 1
+        WR_EN OFFSET(2) NUMBITS(1) []
+    ]
+];
+
+const GLOBALSEC_BASE_ADDR: u32 = 0x4009_0000;
+const GLOBALSEC_REGISTERS: StaticRef<Registers> =
+    unsafe { StaticRef::new(GLOBALSEC_BASE_ADDR as *const Registers) };
+
+pub static mut GLOBALSEC: GlobalSecHardware = GlobalSecHardware::new(GLOBALSEC_REGISTERS);
+
+pub struct Segments {
+    pub ro_a: SegmentInfo,
+    pub ro_b: SegmentInfo,
+    pub rw_a: SegmentInfo,
+    pub rw_b: SegmentInfo,
+}
+
+/// GlobalSec
+pub struct GlobalSecHardware {
+    registers: StaticRef<Registers>,
+    runtime_segment_info: RuntimeSegmentInfo,
+}
+
+impl GlobalSecHardware {
+    const fn new(base_addr: StaticRef<Registers>) -> GlobalSecHardware {
+        let globalsec = GlobalSecHardware {
+            registers: base_addr,
+            runtime_segment_info: UNKNOWN_RUNTIME_SEGMENT_INFO,
+        };
+
+        globalsec
```

Comment:
You don't need the `globalsec` local variable here, just return the new struct directly.

Reply:
No idea what I was thinking there. Fixed.